core/src/test/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorterSuite.java

```diff
@@ -101,7 +101,7 @@ public OutputStream apply(OutputStream stream) {
   public void setUp() {
     MockitoAnnotations.initMocks(this);
     sparkConf = new SparkConf();
-    tempDir = new File(Utils.createTempDir$default$1());
+    tempDir = Utils.createTempDir(System.getProperty("java.io.tmpdir"), "unsafe-test");
```

Comment:
In this case no; setUp and tearDown are per-test, so each test gets a separate temp dir.

I'll take a look at the other tests. Might be worth it to add the temp dir helper method in some common java class.